From: Timo Hoenig <thoenig@suse.de>
Date: Mon, 23 Oct 2006 15:35:17 +0000 (-0400)
Subject: don't call dbus_connection_close() for some of our tools
X-Git-Url: http://gitweb.freedesktop.org/?p=hal.git;a=commitdiff;h=c85e02d58d6503b494bb4c8584a7c44ac6923208

don't call dbus_connection_close() for some of our tools

lshal and hal-device should not call dbus_connection_close() as they do
not have a private connection to the bus.

This is not critical. However, with D-Bus 0.94+ it will spawn ugly error
messages.  Note: It is safe to apply this patch for HAL when using older
versions of D-Bus, too.
---

--- a/tools/hal-device.c
+++ b/tools/hal-device.c
@@ -161,7 +161,6 @@ int main(int argc, char **argv)
 
 	libhal_ctx_shutdown(hal_ctx, &error);
 	libhal_ctx_free(hal_ctx);
-	dbus_connection_close(conn);
 	dbus_connection_unref(conn);
 	dbus_error_free(&error);
 
--- a/tools/lshal.c
+++ b/tools/lshal.c
@@ -723,7 +723,6 @@ main (int argc, char *argv[])
 		LIBHAL_FREE_DBUS_ERROR (&error);
 	libhal_ctx_free (hal_ctx);
 
-	dbus_connection_close (conn);
 	dbus_connection_unref (conn);
 
 	if (show_device)

